blog.md

```diff
@@ -0,0 +1,166 @@
+
+![](./images/spektate-in-progress-deployments.png)
+
+### Capturing Data
+
+Spektate uses a storage table to capture the details of every deployment attempt in the GitOps process. When the first code change is made to the source code, it's associated with a docker image, which creates a change in the HLD and eventually makes its way to the manifest when approved. All these details are captured in a storage table using bash scripts inserted into the CI/CD pipelines by bedrock-cli. For this Trivia app, when I used bedrock-cli to configure the pipelines, it created the bash scripts automatically along with a storage container in my Azure subscription. 
+
+### Extendible
+
+Spektate can be easily extended to work with any other storage tables, but currently we've only added support for Azure storage table. If you would like to use it with another CI/CD orchestrator other than Azure DevOps, we've support for GitHub Actions coming soon, and we can apply the same idea to any orchestrator as well. 
+
+### Deployment
+
+Spektate can be deployed in a Kubernetes cluster by using the [helm chart](https://github.com/microsoft/spektate/tree/master/chart) for Spektate. 
+
+### Security
+
+Spektate does not access your cluster directly, it only needs access to your pipelines if they are private and your repositories to gather information. The keys are stored securely using Kubernetes secrets. 
+
+# What next?
+
+We're working to add support for Github Actions into Spektate and make Spektate less dependent on external APIs by capturing all necessary data into the storage table. This will enable us to run Spektate on any ecosystem. We're also looking to make Spektate columns and table more customizable. 
+
+The work in this area is ongoing and there's a long journey ahead - feel free to reach out to us with feedback on [Github](https://github.com/microsoft/spektate) or [Twitter](https://twitter.com/samiyaakhtar). 
+
+# Useful Links
+
+Our presentation at Kubecon 2020 is available on [YouTube](https://www.youtube.com/watch?v=JfQvAtsZP7Y). 
+
+[Spektate](https://github.com/microsoft/spektate)
+
+[Bedrock](https://github.com/Microsoft/bedrock)
+- [5 Minute GitOps Pipeline with bedrock](https://github.com/microsoft/bedrock/blob/master/docs/gitops-quickstart.md)
+- [Automated Kubernetes deployments with Bedrock](https://docs.microsoft.com/en-us/azure/architecture/example-scenario/bedrock/bedrock-automated-deployments)
+
+[Fabrikate](https://github.com/microsoft/fabrikate)
+
+[Bedrock-cli](https://github.com/Microsoft/bedrock-cli)
+
+Trivia sample repositories: 
+- [Source code](https://dev.azure.com/epicstuff/hellobedrock/_git/hello-world-full-stack).
+- [HLD](https://dev.azure.com/epicstuff/hellobedrock/_git/hello-world-full-stack-hld). 
+- [Manifest](https://dev.azure.com/epicstuff/hellobedrock/_git/hello-world-full-stack-manifest)
+
+TODO: Host this somewhere else: [A sample dashboard](http://40.64.74.69:5000/)
```

Comment:
It might be difficult to host this unless we have dedicated hardware. Using ACI might have slow startup times.

Reply:
i just wanted to host it on something like `spektatedemo.azurewebsites.net` using a web service, but I think what we have now is fine too.